### PR TITLE
Updating Identity Engine > Embedded Authentication content for Node.js SDK Password and Phone factor use case

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Adding the SDK password and phone factor use case for Nodejs language selector in this Identity Engine guide: [Sign in with password and phone factors](https://developer.okta.com/docs/guides/oie-embedded-sdk-use-cases/aspnet/oie-embedded-sdk-use-case-sign-in-pwd-phone/)
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-408181](https://oktainc.atlassian.net/browse/OKTA-408181)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md [hunk 1]
```diff
@@ -1 +1,19 @@
...
+![Sign in with email and phone 1](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-nodejs-seq-1.png
+ "Sign in with email and phone 1")
+
+</div>
+
+### Sequence flow (part 2)
+
+<div class="common-image-format">
+
+![Sign in with email and phone 2](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-nodejs-seq-2.png
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md:14]: update alt-text description

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md [hunk 2]
```diff
@@ -1 +1,19 @@
...
+ "Sign in with email and phone 2")
+
+</div>
+
+>**NOTE:** The SDK sample application does not support this use case.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md]: Can you be specific and indicate that the Node.js sample app doesn't support this use case (or maybe have a link to the sample app?).

**Review by vanngo-okta — COMMENTED** [review]
Wow, this was a lot to write if the sample app didn't have this use case! Kudos to you!
A note about the sequence diagrams: can you update the phrases on the arrows to be consistent? Either command (such as "Enter credentials", "Return response") or an action done by the actor (such as "Clicks continue", "Returns response"). I'm only commenting on this because I made the same mistake and Chris pointed it out :-) I ended up using all command phrases.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 3]
```diff
@@ -1 +1,95 @@
...
\ No newline at end of file
+## Integration steps
+
+### Step 1: Build a sign-in page on the client and authenticate user credentials
+
+Build a sign-in page that captures the user's name and password, as shown in the following example.
+
+<div class="common-image-format">
+
+![Sign-in page](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-sign-on-screenshot-sign-in.png
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md:9]: Nice catch. Not used to alt-text in dev docs.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 4]
```diff
@@ -1 +1,95 @@
...
+ "Sign-in page")
+
+</div>
+
+When the user initiates the sign-in process, your app needs to create a new `OktaAuth` object and set its `username` and `password` properties to the values entered by the user. Send this object to `idx.authenticate` to start the authentication process. See [idx.Authenticate](https://github.com/okta/okta-auth-js/blob/master/docs/idx.md#idxauthenticate) for more information. This call begins the multi-factor authentication flow.
+
+If the username and password are valid, `idx.authenticate` should return a status of `Idx.Status:PENDING`. This status indicates that an additional factor needs to be verified before the sign in. In addition to the status, the `nextStep` field, included in the response, identifies the input parameters of the next step, in this case an authenticator of type phone:
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md:16]: Thanks. All good fixes.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 5]
```diff
@@ -1 +1,95 @@
...
+    inputs, // [{ name: 'authenticator', ... }]
+    options // [{ name: 'phone', ... }, ...]
+  }
+```
+
+The user should be redirected to an authenticator list page that displays the phone factor as an authenticator to be verified.
+
+### Step 2: Show the phone factor in the authenticator list
+
+The next step is to show the phone factor in an authenticator list page. If not already done, this page needs to be built out and display the list of authenticators from the previous step. In this use case, only the **phone** factor will be displayed, as shown in the following sample screenshot.
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md]: Yes, repeated content here. I've removed that bit.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 6]
```diff
@@ -1 +1,95 @@
...
+
+<div class="common-image-format">
+
+![Verify phone](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-screen-verify-phone.png
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md]: Removed the image.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 7]
```diff
@@ -1 +1,95 @@
...
+  nextStep: {
+    inputs, // [{ phone: '+#######', ... }]
+  }
+```
+
+The next step is to redirect the user to a page to enter in the phone number.
+
+### Step 3: Build the phone number entry page
+
+Build the phone number entry page that accepts the phone number that the user will enroll and verify.
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md]: I recast this part as follows (my SDK app accepts phone number and verification method on one page):

Build the phone number entry page that accepts the phone number and phone verification method that the user will use for enrollment.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 8]
```diff
@@ -1 +1,95 @@
...
+
+<div class="common-image-format">
+
+![Verify phone](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-self-serv-screen-verify-phone-num.png
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md]: Removed this image, too

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 9]
```diff
@@ -1 +1,95 @@
...
+ "Verify phone")
+
+</div>
+
+> **Note:** The SDK requires that the phone number follows the `+#######` format, which starts with a plus (+) sign. See [Data Requirements - Phone number](/docs/guides/oie-embedded-sdk-common/nodejs/main/#phone-number).
+
+When the user enters a phone number, send the user to another page that selects the phone verification method, either SMS or voice verification. After a user clicks the send code through the SMS button, a call to `idx.authenticate` is made that passes in the following values:
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md]: Recast this sentence based on SME review

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 10]
```diff
@@ -1 +1,95 @@
...
+<div class="common-image-format">
+
+![Verify phone](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-self-serv-screen-verify-phone-code.png
+ "Verify phone")
+
+</div>
+
+The next step is to call `idx.authenticate` passing in the verification code `({ verification: 'xxx'})`.
+
+The next step is to handle the response from `idx.authenticate`. If the phone code was valid, the method should return a status of `Idx.Status:SUCCESS` with tokens. This status signifies that there are no more factors waiting to be enrolled and verified. If the steps described in [Set up your Okta org (for multi-factor use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-multifactor-use-cases) were properly followed, the user has successfully signed in and should be sent to the default sign-in home page.
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md]: Yes, recast this bit after SME and my review.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md [hunk 11]
```diff
@@ -1 +1,19 @@
-## Summary of step
\ No newline at end of file
+### Sequence flow (part 1)
+
+<div class="common-image-format">
+
+![Sign in with email and phone 1](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-nodejs-seq-1.png
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md:5]: removed image

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md (hunk 2 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md]: Thanks, removed this note as it was in error

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 12]
```diff
@@ -1 +1,95 @@
...
+  }
+```
+
+### Step 4: Build or reuse the phone verification code page
+
+Build a page that accepts the code sent to your phone number through SMS. Depending on your implementation, this page can be the same page that verifies the email code or a different page.
+
+<div class="common-image-format">
+
+![Verify phone](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-self-serv-screen-verify-phone-code.png
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md]: Removed image.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 13]
```diff
@@ -1 +1,74 @@
...
\ No newline at end of file
+## Integration steps
+
+### Step 1: Build a sign-in page on the client and authenticate user credentials
+
+Build a sign-in page that captures the username and password, as shown in the following example.
+
+<div class="common-image-format">
+
+![Sign-in page](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-sign-on-screenshot-sign-in.png
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md:9]:
```suggestion
![Sign-in page screenshot where user enters username and password for authentication.](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-sign-on-screenshot-sign-in.png)
```

Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md:9]: Chris pointed out that we got our alt-text in the wrong location. I've updated the style guide to reflect this.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 14]
```diff
@@ -1 +1,74 @@
...
+ "Sign-in page screenshot where user enters username and password for authentication.")
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md:10]:
```suggestion
```

Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md:10]: Remove this line. Title is not required if the alt-text describes the image.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md [hunk 15]
```diff
@@ -1 +1,74 @@
...
+}
+```
+
+### Step 4: Build or reuse the phone verification code page
+
+Build a page that accepts the code sent to the user's phone number through SMS. Depending on your implementation, this page can be the same page that verifies the email code or a different page.
+
+The next step is to call `idx.authenticate` passing in the verification code `{ verification: 'xxx'}`.
+
+Then, handle the response from `idx.authenticate`. If the phone code was valid, the method returns a status of `Idx.Status:SUCCESS` with tokens. This status signifies that there are no more factors waiting to be enrolled and verified. If the steps described in [Set up your Okta org (for multifactor use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-multifactor-use-cases) were properly followed, the user has successfully signed in and should be sent to the default sign-in home page.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/integrationsteps.md:70]:
```suggestion
Then, handle the response from `idx.authenticate`. If the phone code was valid, the method returns a status of `Idx.Status:SUCCESS` with tokens. This status signifies that there are no more factors waiting to be enrolled and verified. If the steps described in [Set up your Okta org for multifactor use cases](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-multifactor-use-cases) were properly followed, the user has successfully signed in and is sent to the default sign-in home page.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md [hunk 16]
```diff
@@ -1 +1,17 @@
...
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md:5]:
```suggestion
![Displays part 1 of the sign in with email and phone flow sequence steps.](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-nodejs-seq-1.png)
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md [hunk 17]
```diff
@@ -1 +1,17 @@
...
+ "Sign in with email and phone 1")
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md:6]:
```suggestion
```

Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md:6]: title not required

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md [hunk 18]
```diff
@@ -1 +1,17 @@
...
+
+</div>
+
+### Sequence flow (part 2)
+
+<div class="common-image-format">
+
+![Sign in with email and phone 2](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-nodejs-seq-2.png
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md:14]:
```suggestion
![Displays part 2 of the sign in with email and phone flow sequence steps.](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-sign-in-pwd-phone-nodejs-seq-2.png)
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md [hunk 19]
```diff
@@ -1 +1,17 @@
...
+ "Sign in with email and phone 2")
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md:15]:
```suggestion
```

Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-pwd-phone/nodejs/summaryofsteps.md:15]: title not required

**Review by vanngo-okta — APPROVED** [review]
Some nits. I'm approving since it LGTM! 🚀

**Pull request merged**